torch/csrc/utils/tensor_types.cpp

```diff
@@ -14,46 +15,46 @@ using namespace at;
 
 namespace torch { namespace utils {
 
-static const char* backend_to_string(const at::Type& type) {
-  switch (type.backend()) {
+static const char* backend_to_string(const at::Backend& backend) {
+  switch (backend) {
     case at::Backend::CPU: return "torch";
     case at::Backend::CUDA: return "torch.cuda";
     case at::Backend::SparseCPU: return "torch.sparse";
     case at::Backend::SparseCUDA: return "torch.cuda.sparse";
     // We split complex into its own backend, but keeping it the same here for now
     case at::Backend::ComplexCPU: return "torch";
     case at::Backend::ComplexCUDA: return "torch.cuda";
-    default: AT_ERROR("Unimplemented backend ", type.backend());
+    default: AT_ERROR("Unimplemented backend ", backend);
   }
 }
 
-std::string type_to_string(const at::Type& type, const ScalarType scalar_type) {
+std::string type_to_string(const at::DeprecatedTypeProperties& type) {
   std::ostringstream ss;
-  ss << backend_to_string(type) << "." << toString(scalar_type) << "Tensor";
+  ss << backend_to_string(type.backend()) << "." << toString(type.scalarType()) << "Tensor";
   return ss.str();
 }
 
-std::pair<at::Type*, at::ScalarType> type_from_string(const std::string& str) {
+at::DeprecatedTypeProperties* type_from_string(const std::string& str) {
   static std::string cuda_prefix("torch.cuda.");
   static std::once_flag cpu_once;
   static std::once_flag cuda_once;
-  static std::unordered_map<std::string, std::pair<at::Type*, at::ScalarType>> cpu_map;
-  static std::unordered_map<std::string, std::pair<at::Type*, at::ScalarType>> cuda_map;
+  static std::unordered_map<std::string, at::DeprecatedTypeProperties*> cpu_map;
+  static std::unordered_map<std::string, at::DeprecatedTypeProperties*> cuda_map;
 
-  const std::unordered_map<std::string, std::pair<at::Type*, at::ScalarType>>* map = nullptr;
+  const std::unordered_map<std::string, at::DeprecatedTypeProperties*>* map = nullptr;
 
   if (str == "torch.Tensor") {
-    return std::make_pair(&torch::tensors::get_default_tensor_type(),
-                          torch::tensors::get_default_scalar_type());
+    auto backend = tensorTypeIdToBackend(torch::tensors::get_default_tensor_type_id());
+    auto scalar_type = torch::tensors::get_default_scalar_type();
+    return &getNonVariableDeprecatedTypeProperties(backend, scalar_type);
   }
 
   if (std::mismatch(cuda_prefix.begin(), cuda_prefix.end(), str.begin()).first == cuda_prefix.end()) {
     // torch.cuda. is prefix of str
     std::call_once(cuda_once, []() {
       for (auto type : autograd::VariableType::allCUDATypes()) {
         for (int s = 0; s < static_cast<int>(ScalarType::NumOptions); s++) {
```

Comment:
wait, do you still need to loop over ScalarTypes here?

Reply:
Yeah, we still need to return Backend, ScalarType pair.

Reply:
but you are never using `s`?

Reply:
Oh I see what you mean